data/filters/youtube-channel.yaml

```diff
@@ -0,0 +1,38 @@
+title: Hide Youtube videos by channel
+params:
+  - name: keywords
+    description: Channels to search for
+    type: list
+    default: [ "ChannelURL" ]
+tags:
+  - youtube
+template: |
+  {{#each keywords}}
+  www.youtube.com##ytd-browse ytd-grid-video-renderer:has([href*="{{ . }}" i])
+  www.youtube.com##ytd-browse ytd-rich-item-renderer:has([href*="{{ . }}" i])
+  www.youtube.com##ytd-search ytd-video-renderer:has([href*="{{ . }}" i])
+  www.youtube.com##ytd-search ytd-channel-renderer:has([href*="{{ . }}" i])
+  www.youtube.com##ytd-search #channel-name:has-text(/{{ . }}/i):upward(ytd-secondary-search-container-renderer)
+  www.youtube.com##ytd-watch-next-secondary-results-renderer #channel-name:has-text(/{{ . }}/i):upward(ytd-compact-video-renderer)
```

Comment:
Unfortunately, these two rules would require a different parameter than the other rules, as these are matching on the channel name instead of the URL.

I think we should either:

- drop these rules, and only focus on where we can efficiently match the channel URL (preferred, because I agree with you that matching the channel name is too brittle)
- move all the rules to text matching on the channel name for consistency

AFAIK, the sidebar suggestions to respect the "don't recommend channel" setting, so it's probably a low priority to implement this?

Reply:
I think an important question here is: How many people use YT without being logged in?
If there are a lot of people who do so, I think I prefer your second option to match the text in stead of the URL